app/actions/briefActions.js

```diff
@@ -0,0 +1,88 @@
+import {
+  DATA_IS_LOADING,
+  BRIEF_INFO_FETCH_DATA_SUCCESS,
+  BRIEF_INFO_HAS_ERRORED,
+  BRIEF_RESPONSE_SUCCESS,
+  BRIEF_RESPONSE_FAILURE,
+  BRIEF_RESPONSE_DUPLICATE_FAILURE
+} from '../constants/constants'
+
+import dmapi from '../services/apiClient'
+
+export function handleDataLoading(bool) {
+  return {
+    type: DATA_IS_LOADING,
+    isLoading: bool
+  }
+}
+
+export function handleBriefInfoSuccess(response) {
+  return {
+    type: BRIEF_INFO_FETCH_DATA_SUCCESS,
+    brief: response.data
+  }
+}
+
+export function handleBriefInfoFailure(response) {
```

Comment:
If you look at master I've managed to refactor these specific failure cases into a generic error